feat(linter): add `vue/no-import-compiler-macros` rule

Sysix opened the pull request:
waiting for the PR result of https://github.com/vuejs/eslint-plugin-vue/pull/2938

related #11440 
https://eslint.vuejs.org/rules/no-import-compiler-macros

Sysix commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/14335?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#14335** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/14335?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/14335?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#14307** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/14307?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#14306** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/14306?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#14305** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/14305?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=14335).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Sysix commented on crates/oxc_linter/src/rules/vue/no_import_compiler_macros.rs:
```diff
@@ -0,0 +1,278 @@
+                            import_decl.span.end,
+                        ));
+                        let next_comma_index = part_source.find(',').unwrap_or_default();
+                        fixer.delete_range(Span::new(
+                            import_specifier.span.start,
+                            import_specifier.span.end + next_comma_index as u32 + 1,
```
@camc314 good enough for me. What do you think?

camc314 commented:
### Merge activity

* **Oct 7, 4:03 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Sysix committed:
feat(linter): add `vue/no-import-compiler-macros` rule (#14335)

waiting for the PR result of https://github.com/vuejs/eslint-plugin-vue/pull/2938

related #11440
https://eslint.vuejs.org/rules/no-import-compiler-macros
